### DISCUSSION TITLE
How can I make files name same as original filename??

### DISCUSSION THREAD

**Discussion by deletedaccnt260427** [Comfy-Org/ComfyUI#4677] [Q&A]
![1](https://github.com/user-attachments/assets/7eaa0a5d-9c94-420d-809c-2a01cf145aae)
![2](https://github.com/user-attachments/assets/14f22062-6a0c-4286-bb43-05426f8473ff)
![3](https://github.com/user-attachments/assets/03d78cf5-f29c-4e8e-a38f-79ffdebe1486)

The workflow involves upscaling and saving the images. Is there a way to save the upscaled images with the same names as the original ones?

**Comment by fengfengwu** [comment]
I also want to know.

**Comment by Mubumbutu** [comment]
I would also like to know, it would make it easier to create hd packs for ps2 games

**Comment by Rogala** [comment]
Install Node [ComfyUI_yanc](https://github.com/ALatentPlace/ComfyUI_yanc) select a module to upload a single file or batch processing from a folder and a module to save its modules and link them.
![image](https://github.com/user-attachments/assets/8d2a4b35-771f-4c75-85b0-893280010c0c)

**Comment by Mubumbutu** [comment]
Thank you, Rogala, it’s nice to create single images this way. However, is there another method? In this case, I have to generate each image individually, and with around 4/5 thousand images, it requires quite a bit of clicking. Additionally, when generating from a folder, this node doesn’t have a mask.

**Reply by Rogala** [reply]
About Noda with a mask for such a task I will not say, but I do not understand why click on one photo, just do put the necessary photos in a folder and run a continuous task, it will search all the pictures itself.
![image](https://github.com/user-attachments/assets/7ee57ec2-99ae-48cd-98b5-94208a0b0d40)

**Reply by Mubumbutu** [reply]
It works with the 'Queue (instant)' option, thank you for your time and knowledge, it will definitely come in handy for other things in the future. However, without a mask, it creates an image with a black background instead of alpha, which is unacceptable for textures. Does anyone know how to fix this? Since it can be done without assigning names automatically, there must be a way to do it?

**Reply by Rogala** [reply]
Check these nodes, but I haven't checked them myself.
![node](https://github.com/user-attachments/assets/033cc64c-37fb-4a63-bda2-c17633169459)
